### ISSUE TITLE
Invalid lint error when `COPY --from=<target>` is another image

### ISSUE THREAD

**Issue by zx8** [hadolint/hadolint#421]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

There should be no suggestion.

### Actual behavior

```
Dockerfile:3 DL3022 COPY --from should reference a previously defined FROM alias
```

### Steps to reproduce the behavior

**Output of `hadolint --version`**

```bash
Haskell Dockerfile Linter 1.17.6-no-git
```

**Dockerfile** (if relevant)

```Dockerfile
FROM scratch

COPY --from=docker:19.03.8 /usr/local/bin/docker /usr/local/bin/
```

**Comment by SuperSandro2000** [comment]
It wants the following:
```
FROM docker:19.03.8 as docker

FROM scratch

COPY --from=docker /usr/local/bin/docker /usr/local/bin/
```

**Comment by lorenzo** [comment]
Yeah, this is on purpose. It is more clear to import from a stage than from a magic string that may mean an image. That way you get better errors in your builds

**Comment by gomex** [comment]
But if we need to import a binary from the external image? I don't think this should be a bad practice.

**Comment by joe-sonrichard** [comment]
It's definitely part of Docker sytnax: https://docs.docker.com/develop/develop-images/multistage-build/#use-an-external-image-as-a-stage I am also running across this error.

**Comment by lorenzo** [comment]
Hey, the rule is on purpose. This is the reason is that we believe is better to "import" using `FROM` as you would do in a regular programming language. This lets us verify that the image is correctly alias ahead of time and that it does not clash with other aliases.

You may also ignore the rule if you find it inconvenient.
